ruby/tests/basic.rb

```diff
@@ -9,270 +14,90 @@
 
-      assert_equal 2, m.number
-      assert_equal [10, 20, 30], m.repeated_number
-
-      assert_not_nil m.child_msg
-      assert_not_nil m.child_msg.sub_child
-      assert_equal m.child_msg.sub_child.optional_int32, 1
-
-      assert_not_nil m.repeated_msg
-      assert_equal 1, m.repeated_msg.length
-      assert_equal 3, m.repeated_msg.first.sub_child.optional_int32
-    end
-
-    def test_inspect
-      m = TestMessage.new(:optional_int32 => -42,
-                          :optional_enum => :A,
-                          :optional_msg => TestMessage2.new,
-                          :repeated_string => ["hello", "there", "world"])
-      expected = '<BasicTest::TestMessage: optional_int32: -42, optional_int64: 0, optional_uint32: 0, optional_uint64: 0, optional_bool: false, optional_float: 0.0, optional_double: 0.0, optional_string: "", optional_bytes: "", optional_msg: <BasicTest::TestMessage2: foo: 0>, optional_enum: :A, repeated_int32: [], repeated_int64: [], repeated_uint32: [], repeated_uint64: [], repeated_bool: [], repeated_float: [], repeated_double: [], repeated_string: ["hello", "there", "world"], repeated_bytes: [], repeated_msg: [], repeated_enum: []>'
-      assert_equal expected, m.inspect
-    end
-
-    def test_hash
-      m1 = TestMessage.new(:optional_int32 => 42)
-      m2 = TestMessage.new(:optional_int32 => 102, repeated_string: ['please', 'work', 'ok?'])
-      m3 = TestMessage.new(:optional_int32 => 102, repeated_string: ['please', 'work', 'ok?'])
-      assert m1.hash != 0
-      assert m2.hash != 0
-      assert m3.hash != 0
-      # relying on the randomness here -- if hash function changes and we are
-      # unlucky enough to get a collision, then change the values above.
-      assert m1.hash != m2.hash
-      assert_equal m2.hash, m3.hash
-    end
-
-    def test_unknown_field_errors
-      e = assert_raise NoMethodError do
-        TestMessage.new.hello
-      end
-      assert_match(/hello/, e.message)
+      m.optional_string = "foo bar"
+      assert_equal "foo bar", m.optional_string
+      m.clear_optional_string
+      assert_equal "", m.optional_string
 
-      e = assert_raise NoMethodError do
-        TestMessage.new.hello = "world"
+      m.optional_msg = TestMessage2.new(:foo => 42)
+      assert_equal TestMessage2.new(:foo => 42), m.optional_msg
+      assert_true m.has_optional_msg?
+      m.clear_optional_msg
+      assert_equal nil, m.optional_msg
+      assert_false m.has_optional_msg?
+
+      m.repeated_int32.push(1)
+      assert_equal [1], m.repeated_int32
+      m.clear_repeated_int32
+      assert_equal [], m.repeated_int32
+
+      m = OneofMessage.new
+      assert_raise RuntimeError do
+        m.clear_my_oneof
       end
-      assert_match(/hello/, e.message)
+      m.a = "foo"
+      assert_equal "foo", m.a
+      assert_true m.has_my_oneof?
+      m.clear_a
+      assert_false m.has_my_oneof?
     end
 
+
     def test_initialization_map_errors
```

Comment:
Why this hasn't been moved to common test?

Reply:
This PR doesn't add support for maps for proto2 (plan to do it in a followup). And since this test uses MapMessage, I have kept it separate for now. Will put it in `CommonTests` module once I add map support in proto2.